src/Symfony/Component/HttpFoundation/HeaderBag.php

```diff
@@ -121,7 +121,7 @@ public function get($key, $default = null, $first = true)
         }
 
         if ($first) {
-            return \count($headers[$key]) ? (string) $headers[$key][0] : $default;
+            return \count($headers[$key]) && null !== $headers[$key][0] ? (string) $headers[$key][0] : $default;
```

Comment:
Imagine $default is not null: this should still return null when the key exists. Can you please fix this and update the test case to catch this?

Reply:
Okay, I've actually thought null would be interpreted more like as if it is not set, hence this implementation. But I'll change it.